torch/nested/_internal/ops.py

```diff
@@ -1586,6 +1586,36 @@ def _nested_get_max_seqlen(func, *args, **kwargs):
     return inp._metadata_cache.get("max_seqlen", None)
 
 
+# If a section of the Nested Tensor is fully masked out we still retain the section with a length of 0
+@register_jagged_func(torch.ops.aten.masked_select.default, "self: jt, mask: any")
+def masked_select_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    inp = new_kwargs.pop("input")
+    mask = new_kwargs.pop("mask")
+
+    mask_values = None
+    if inp.ndim > 2:
+        raise RuntimeError("masked_select only support 2-D selections currently")
+    elif inp.shape == mask.shape:
+        mask_values = mask.values()
```

Comment:
nit: no reason to have to create the mask_values variable anymore since we only support one case

Reply:
Makes sense, btw the reduction to one case was due to the testing framework limitations of size comparisons (and we didn't need the extra case for an immediate use case). I can add it in a later PR if we need that case or want to support it